### PR TITLE
Default assignments on classroom join

### PR DESCRIPTION
**Pull request by zwolf**
For "custom" (Wildcam-style, custom assignment) programs, teachers create assignments before or after a classroom is joined and then assign them to students. However, for non-custom (i2a) programs, the order of operations will look more like this:

1) Teacher creates classroom
2) Default assignments are automatically generated by the i2a frontend.
3) Stutent attempts to join classroom and is automatically assigned each of the classroom's assignments.

This PR adds that check, based on the program's `custom` attribute, and assigns them if it is false. Programs are now required by all classroom requests and are an included association in the classroom factory.

### CODE REVIEW

File: spec/operations/classrooms/join_spec.rb [hunk 1]
```diff
@@ -29,4 +29,29 @@
...
   end
+
+  describe "assignment assignment" do
+    describe "custom programs" do
+      let(:custom_program) { create(:program, custom: true) }
+      let(:custom_classroom) { create(:classroom, program: custom_program) }
+      let!(:custom_assignment) { create(:assignment, classroom: custom_classroom) }
+
+      it "doesn't affect assignments" do
+        expect{
```
[the remark this answers is not in the document]
Comment (zwolf) [reply spec/operations/classrooms/join_spec.rb]: seems like a known rubocop bug (https://github.com/bbatsov/rubocop/issues/4198), updating might fix.

**Comment by zwolf** [comment]
I incorporated a couple older, unmerged PRs related to classroom requests requiring programs. Should be ready for review now, though, @marten.

**Comment by srallen** [comment]
Just so you all know the timeline for this, we're tentatively planning on deploying the education api to production the first week of December while @zwolf and I are in Oxford. I'm out the third week of Nov (in about a week and a half) and I'm depending on this feature to finish the remaining I2A front end features. Getting this reviewed sooner than later would be very helpful. Thanks.

**Comment by zwolf** [comment]
@marten mind taking a quick look at this?

**Pull request merged**